test/unit/resource_manager/test_resource_manager.py

```diff
@@ -0,0 +1,94 @@
+import base64
+from pathlib import Path
+
+import pytest
+
+from voicevox_engine.resource_manager import ResourceManager, ResourceManagerError
+
+with_filemap_dir = Path(__file__).parent / "with_filemap"
+without_filemap_dir = Path(__file__).parent / "without_filemap"
+
+
+def b64encode_str(s: bytes) -> str:
+    return base64.b64encode(s).decode("utf-8")
+
+
+def _assert_resource(manager: ResourceManager, test_path: Path) -> None:
+    """
+    `test_path`で指定したファイルが正しくbase64とハッシュが取得できるか確認する
+    また、取得したハッシュから取得したファイルから同じバイト列が取得できるか確認する
+    """
+    test_bytes = test_path.read_bytes()
+
+    assert manager.resource_str(test_path, "base64") == b64encode_str(test_bytes)
+
+    test_filehash = manager.resource_str(test_path, "hash")
+    test_res_path = manager.resource_path(test_filehash)
+    assert test_res_path is not None
+    assert test_res_path.read_bytes() == test_bytes
+
+
+def test_with_filemap() -> None:
+    """
+    "filemap.json"があるディレクトリでのテスト
+    fimemapの生成方法
+    `python build_util/generate_filemap.py --target_dir test/unit/resource_manager/with_filemap`
+    """
+    manager = ResourceManager(False)
+    manager.register_dir(with_filemap_dir)
+
+    png_path = with_filemap_dir / "dummy.png"
+    _assert_resource(manager, png_path)
+
+    wav_path = with_filemap_dir / "dummy.wav"
+    _assert_resource(manager, wav_path)
+
+    # 同じバイナリがある場合のテスト
+    same_wav_path = with_filemap_dir / "dummy_same_binary.wav"
+    assert wav_path.read_bytes() == same_wav_path.read_bytes()
+    _assert_resource(manager, same_wav_path)
+
+    # テキストは通常エンコードせずにJSONに含めるためResourceManagerでは管理しない
+    txt_path = with_filemap_dir / "dummy.txt"
+    with pytest.raises(ResourceManagerError) as _:
+        manager.resource_str(txt_path, "base64")
+    with pytest.raises(ResourceManagerError) as _:
+        manager.resource_str(txt_path, "hash")
+
+    # 登録されていないハッシュが渡された場合Noneを返す
+    assert manager.resource_path("NOT_EXIST_HASH") is None
+
+
+def test_without_filemap_when_production() -> None:
+    """
+    "create_filemap_if_not_exist"がFalseで"filemap.json"が無い場合エラーにする(製品版を想定)
+    """
+    manager = ResourceManager(False)
+    with pytest.raises(ResourceManagerError) as _:
+        manager.register_dir(without_filemap_dir)
+
+
+def test_without_filemap() -> None:
+    """
+    "create_filemap_if_not_exist"がTrueで"filemap.json"が無い場合エラーにする(開発時を想定)
+    """
+    manager = ResourceManager(True)
+    manager.register_dir(without_filemap_dir)
+
+    png_path = without_filemap_dir / "dummy.png"
+    _assert_resource(manager, png_path)
+
+    wav_path = without_filemap_dir / "dummy.wav"
+    _assert_resource(manager, wav_path)
+
+    # 同じバイナリがある場合のテスト
+    same_wav_path = without_filemap_dir / "dummy_same_binary.wav"
+    assert wav_path.read_bytes() == same_wav_path.read_bytes()
+    _assert_resource(manager, same_wav_path)
+
+    # "filemap.json"がない場合、全てのファイルが公開される
```

Comment:
公開されるかどうかはResourceManagerの範疇じゃなさそう。
あとコメントも間違えてそう。（filemap.jsonが無い場合、ではない）

こうですかね。
```suggestion
    # create_filemap_if_not_existがTrueのときは全てのファイルが管理される
```

Reply:
`filemap.json`があるときはそれを使用するので`create_filemap_if_not_exist`が`True`でも全てのファイルは管理対象になりません。
ResourceManagerの動作を変えますか？

Reply:
あーたしかにです！動作はそのままが良いと思います！

どう案内するか難しいですね。。。
とくにコメントを書かず、`wav_path`の下辺りに移して、それらのファイルの上に`全てのファイルが公開される`と注釈を入れる形が良さそうかもです。

filemapとResourceManagerの仕様をどこかにメモ書きしておきたい気持ちがあるのですが、なんか良い場所ないですかね。。。
`/docs/`･･･････？

Reply:
コメント修正しました。

あととりあえず`docs/ResourceManagerの仕様について.md`を作ってみましたがこんなかんじでいいでしょうか？

Reply:
おー良さそうです！！！
ちょっとそちらにコメントさせていただきます！！